packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -79,14 +77,19 @@ class HotRunner extends ResidentRunner {
              packagesFilePath: packagesFilePath,
              saveCompilationTrace: saveCompilationTrace,
              stayResident: stayResident,
-             ipv6: ipv6);
+             ipv6: ipv6)  {
+    _projectWatcher = ProjectWatcher(
+      projectRootPath,
+      packagesFilePath ?? fs.path.absolute(PackageMap.globalPackagesPath),
```

Comment:
Do you to monitor packages file itself and update(rebuild?) ProjectWatcher if it changes?

Reply:
We have to tear down the runner and restart if packages are updated currently

Reply:
What are the plans for this in this PR?

Reply:
Today, we can't update .packages due to https://github.com/flutter/flutter/issues/29186. Essentially we are producing a massive incremental dill (~40MB for the gallery) and that seems to be crashing in part of the shell.

Also related to https://github.com/dart-lang/sdk/issues/36202, where the vmservice does not respond with any caught exceptions.

New implementation is to detect changes to .packages and warn the user that they might have to restart the runner.